### PR TITLE
add stage protection to spacefile

### PR DESCRIPTION
**Pull request by hensur**
this pr is based on #2 
it will apply a stage protection defined in the spacefile and also export the current status back to the spacefile

### CODE REVIEW

File: cmd/spaces_apply.go [hunk 1]
```diff
@@ -40,18 +42,27 @@ CAUTION: This command can be potentially destructive.`,
 			if stageDef == nil {
 				continue
 			}
+
 			if len(stageDef.VirtualHosts) == 0 {
```
Comment (martin-helmich) [inline cmd/spaces_apply.go]: Drop the `if`?

File: spacefile/generator.go [hunk 2]
```diff
@@ -49,6 +49,8 @@ space "{{ .SpaceDNSLabel }}" {
     //     type = "letsencrypt"
     //   }
     // }
+
+	// protection = "oauth"
```
Comment (martin-helmich) [inline spacefile/generator.go]: Tabs vs. spaces

File: client/spaces/protection.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/mittwald/spacectl/client/errors"
+)
+
+// GetStageProtection returns the current Stage Protection type
+func (c *spacesClient) GetStageProtection(spaceID, stage string) (*StageProtection, error) {
+	var protection StageProtection
```
[the remark this answers is not in the document]
Comment (hensur) [reply client/spaces/protection.go]: yes, it always has a value if a protection exists. If there is no protection, the API will return error 404, I now handle this error code by returning the empty struct.

**Comment by martin-helmich** [comment]
Ouch. Did not realize that `protection` was filled in the subsequent `Get` call.
No idea what I was thinking yesterday; never mind.

**Pull request merged**